client/signup/steps/import-url/index.jsx

```diff
@@ -82,23 +63,93 @@ class ImportURLStepComponent extends Component {
 
 	focusInput = () => invoke( this.inputRef, 'focus' );
 
+	setUrlError = urlValidationMessage => this.setState( { urlValidationMessage }, this.focusInput );
+
 	handleSubmit = event => {
 		event.preventDefault();
 		const isValid = this.validateUrl();
 
 		if ( ! isValid ) {
-			this.focusInput();
 			return;
 		}
 
-		const { urlInputValue, stepName } = this.props;
+		const { stepName, translate, urlInputValue } = this.props;
+
+		this.setState( {
+			isLoading: true,
+			urlValidationMessage: '',
+		} );
 
-		this.props
-			.submitImportUrlStep( {
-				siteUrl: urlInputValue,
-				stepName,
-			} )
-			.then( this.props.goToNextStep );
+		wpcom
+			.undocumented()
+			.isSiteImportable( urlInputValue )
+			.then(
+				( {
+					site_engine: siteEngine,
+					site_favicon: siteFavicon,
+					site_status: siteStatus,
+					site_title: siteTitle,
+					site_url: siteUrl,
+					importer_types: importerTypes,
+				} ) => {
+					if ( ! includes( importerTypes, 'url' ) ) {
+						return this.setUrlError(
+							translate(
+								"That doesn't seem to be a Wix or GoDaddy site. Please check the URL and try again."
+							)
+						);
+					}
+
+					if ( 404 === siteStatus ) {
+						return this.setUrlError(
+							translate( 'That site was not found. Please check the URL and try again.' )
```

Comment:
You should be able to trigger this case with something like `foobarlalala.godaddysites.com` -- GoDaddy resolves the request but returns a 404 status because it's not an actual goDaddy site.

Reply:
Ah so it does, thanks!